src/libraries/Microsoft.Extensions.DependencyModel/ref/Microsoft.Extensions.DependencyModel.cs

```diff
@@ -148,34 +148,34 @@ public partial class RuntimeAssetGroup
+        public RuntimeLibrary(string type, string name, string version, string? hash, System.Collections.Generic.IReadOnlyList<Microsoft.Extensions.DependencyModel.RuntimeAssetGroup> runtimeAssemblyGroups, System.Collections.Generic.IReadOnlyList<Microsoft.Extensions.DependencyModel.RuntimeAssetGroup> nativeLibraryGroups, System.Collections.Generic.IEnumerable<Microsoft.Extensions.DependencyModel.ResourceAssembly> resourceAssemblies, System.Collections.Generic.IEnumerable<Microsoft.Extensions.DependencyModel.Dependency> dependencies, bool serviceable, string? path, string? hashPath) : base (default(string), default(string), default(string), default(string), default(System.Collections.Generic.IEnumerable<Microsoft.Extensions.DependencyModel.Dependency>), default(bool)) { }
+        public RuntimeLibrary(string type, string name, string version, string? hash, System.Collections.Generic.IReadOnlyList<Microsoft.Extensions.DependencyModel.RuntimeAssetGroup> runtimeAssemblyGroups, System.Collections.Generic.IReadOnlyList<Microsoft.Extensions.DependencyModel.RuntimeAssetGroup> nativeLibraryGroups, System.Collections.Generic.IEnumerable<Microsoft.Extensions.DependencyModel.ResourceAssembly> resourceAssemblies, System.Collections.Generic.IEnumerable<Microsoft.Extensions.DependencyModel.Dependency> dependencies, bool serviceable, string? path, string? hashPath, string? runtimeStoreManifestName) : base (default(string), default(string), default(string), default(string), default(System.Collections.Generic.IEnumerable<Microsoft.Extensions.DependencyModel.Dependency>), default(bool)) { }
         public System.Collections.Generic.IReadOnlyList<Microsoft.Extensions.DependencyModel.RuntimeAssetGroup> NativeLibraryGroups { get { throw null; } }
         public System.Collections.Generic.IReadOnlyList<Microsoft.Extensions.DependencyModel.ResourceAssembly> ResourceAssemblies { get { throw null; } }
         public System.Collections.Generic.IReadOnlyList<Microsoft.Extensions.DependencyModel.RuntimeAssetGroup> RuntimeAssemblyGroups { get { throw null; } }
     }
     public partial class TargetInfo
     {
-        public TargetInfo(string framework, string runtime, string runtimeSignature, bool isPortable) { }
+        public TargetInfo(string framework, string? runtime, string? runtimeSignature, bool isPortable) { }
         public string Framework { get { throw null; } }
         public bool IsPortable { get { throw null; } }
-        public string Runtime { get { throw null; } }
-        public string RuntimeSignature { get { throw null; } }
+        public string? Runtime { get { throw null; } }
+        public string? RuntimeSignature { get { throw null; } }
     }
 }
 namespace Microsoft.Extensions.DependencyModel.Resolution
```

Comment:
It looks like below 

`public AppBaseCompilationAssemblyResolver(string basePath) { }` 

should be 

`public AppBaseCompilationAssemblyResolver(string? basePath) { }`

As nothing is checking that string for null.

Reply:
`AppBaseCompilationAssemblyResolver.TryResolveAssemblyPaths` could use some work too:

1. Nothing is checking `CompilationLibrary library` for `null`, but it is dereferenced. So we could start throwing `ArgNullException` when it is null.
2. `List<string> assemblies` should be `List<string>? assemblies`, since the code accepts `null`.
    - But looking at the other `ICompilationAssemblyResolver`, they don't appear to allow `null`. So maybe they all should be updated to check `assemblies` for null, and throw `ArgNull`.